ui/desktop/src/config.ts

```diff
@@ -30,6 +30,56 @@ export type SystemConfig =
       name: string;
     };
 
+// Local storage key for extension configs
+const EXTENSION_CONFIGS_KEY = 'goose_extension_configs';
+
+// Store extension config in localStorage
+const storeExtensionConfig = (config: SystemConfig) => {
+  try {
+    const storedConfigs = localStorage.getItem(EXTENSION_CONFIGS_KEY);
+    const configs = storedConfigs ? JSON.parse(storedConfigs) : [];
+
+    // Check if config already exists (based on cmd for stdio, uri for sse, name for builtin)
+    const configExists = configs.some((stored: SystemConfig) => {
+      if (config.type === 'stdio' && stored.type === 'stdio') {
+        return stored.cmd === config.cmd;
+      } else if (config.type === 'sse' && stored.type === 'sse') {
+        return stored.uri === config.uri;
+      } else if (config.type === 'builtin' && stored.type === 'builtin') {
+        return stored.name === config.name;
+      }
+      return false;
+    });
+
+    if (!configExists) {
+      configs.push(config);
+      localStorage.setItem(EXTENSION_CONFIGS_KEY, JSON.stringify(configs));
+      console.log('Extension config stored successfully');
+    } else {
+      console.log('Extension config already exists');
+    }
+  } catch (error) {
+    console.error('Error storing extension config:', error);
+  }
+};
+
+// Load stored extension configs
+export const loadStoredExtensionConfigs = async (): Promise<void> => {
+  debugger;
+  try {
+    const storedConfigs = localStorage.getItem(EXTENSION_CONFIGS_KEY);
+    if (storedConfigs) {
+      const configs = JSON.parse(storedConfigs);
+      for (const config of configs) {
+        await extendGoosed(config);
```

Comment:
@alexhancock not sure this will work as they are stored as one command vs command + array of args which goose requires